### DISCUSSION TITLE
[QUESTION] Elegoo Saturn Remote printing

### DISCUSSION THREAD

**Discussion by git4unrealnondev** [sn4k3/UVtools#422] [Q&A]
I cannot seem to configure my elegoo saturn with remote printing using UVTools.
I ran an nmap scan and udp ports 53 & 3000 were open and viewable.
Im none too sure how to configure it any.
Im going to try and reverse engineer chitubox calls to my elegoo.
Will report the good word back.

**Comment by sn4k3** [comment]
UVtools sends HTTP requests, those are TCP.
Without clear information on comunication i can't make a profile for it.

You can use Wireshark to have the comunication

**Reply by git4unrealnondev** [reply]
I did some searching and found that my elegoo uses port 3000 over udp.
It looks like mostly standard gcode and output that you would get out of pronterface.
To connect to printer:
nc -u PRINTER_IP 3000

Heres some of the work that I've reverse engineered.: 

M4002 - Get version info
M20   - List files
M27   - Status
M4000 - Where is printer currently.
M6032 'FILE' - Load file
M6030 'FILE' - print file
M3000 - Reads file from memory to netcat?
M33 - Disconnect? / Stop Printing??

**Reply by git4unrealnondev** [reply]
Still need to figure out how to upload files to the printer remotely.

Heres an example session from my setup:
linux@archlinux ~]$ nc -u 192.168.1.33 3000
M4002
ok V4.4.3-Z1_LCDZ
M20
Begin file list
DJIFRAME.ctb 2413346
System Volume Information 0
Slicing Software 0
Elegoo Saturn LCD Light Curable 3D Printer User Manual.pdf 11744972
saturn Machine parameters (en).gcode 7398
ELEGOO SATURN Rook Model for Printing Test 0
End file list
ok L:6
M27
Error:It's not printing now!
ok N:3
M4000
ok B:0/0 X:0.000 Y:0.000 Z:60.000 F:0/256 D:0/0/1 
M6032 'djiframe.ctb'    
ok L:2413346
M6030 'djiframe.ctb'
ok N:5
M4000
ok B:0/0 X:0.000 Y:0.000 Z:2.036 F:256/256 D:128057/2413346/0     
M4000
ok B:0/0 X:0.000 Y:0.000 Z:4.644 F:256/256 D:152816/2413346/0

**Reply by sn4k3** [reply]
To send a file you probably need send a request over UDP and the file bytes.
Using wireshark you can see what chitubox sends when you click on send button

**Reply by git4unrealnondev** [reply]
Yeah it looks like it sends the M28 command with the filename.
M28 DJIFRAME1.ctb
Looks like it sends the file over in binary. 
M29 closes it
I cant seem to get netcat to send files over.
Im using resinator3d to test everything.
